### PR TITLE
Show couuntry name in target lang

### PR DESCRIPTION
**Pull request by harrison**
Shows country name in country selector in the target language.

Fixes: https://github.com/gocardless/splash-pages/issues/72

### CODE REVIEW

**Comment by harrison** [comment]
Review @english?

File: app/components/app/app.js [hunk 1]
```diff
@@ -32,30 +32,32 @@ function setCanonicalHref(props) {
 class App extends React.Component {
   displayName = 'App'
 
-  static contextTypes = {
-    router: React.PropTypes.func,
+  static propTypes = {
```
Comment (english) [inline app/components/app/app.js]: The only change to this file is adding `availableCountryNames` as a `prop` and making it available in its `context`? The diff is really hard to read!

File: app/components/footer/footer.js [hunk 2]
```diff
@@ -9,19 +9,18 @@ import Translation from '../translation/translation';
...
 
-function buildAvailableLocalePages(currentLocale, siteLocales, messages) {
+function buildAvailableLocalePages(currentLocale, siteLocales, availableCountryNames) {
   const availableLocales = Object.keys(siteLocales).map(function(locale) {
     const isActive = currentLocale === locale;
     const path = siteLocales[locale].path;
-    const name = getMessage(messages, `country_names.${locale}`);
-    return {locale, isActive, path, name};
+    const name = availableCountryNames[locale];
+    return {locale, isActive, path, name };
```
Comment (english) [inline app/components/footer/footer.js]: Do you know why `eslint` didn't pick up on the spacing here?

File: app/components/footer/footer.js [hunk 3]
```diff
@@ -33,17 +32,17 @@ class Footer extends React.Component {
...
   }
 
   render() {
-    const { routeName, locales, messages, config, availableLocales } = this.context;
+    const { routeName, locales, config, availableLocales, availableCountryNames } = this.context;
 
     const siteLocales = merge({}, getLocalesForRouteName(homeRoute, availableLocales),
       getLocalesForRouteName(routeName, availableLocales));
-    const availableLocalePages = buildAvailableLocalePages(locales, siteLocales, messages);
+    const availableLocalePages = buildAvailableLocalePages(locales, siteLocales, availableCountryNames);
```
Comment (english) [inline app/components/footer/footer.js]: Could the `siteLocales` assignment get moved into `buildAvailableLocalePages`? Might make it a bit easier to follow, it seems its easy for the `render` methods to get quite complex. Do you agree?

File: app/components/app/app.js (hunk 1 above)
Comment (harrison) [inline app/components/app/app.js]: yeah sorry about that, the order was weird before

**Comment by english** [comment]
Looks good!

Its a shame there's so much code required in order to pass down context from top to bottom.

Am I right in thinking that, with this, the country switcher will only show flags for countries that have corresponding pages for the current path?

**Comment by harrison** [comment]
Yeah, we might be able to have one context wrapper class that both the html document and app can use, will see if that works

**Comment by harrison** [comment]
The country switcher will show all available countries for the site, if a page doesn't exist it links to the home page for that locale

**Pull request merged**